libs/utils/test.py

```diff
@@ -100,4 +102,24 @@ def _experimentsFinalize(cls):
         Code executed after running the experiments
         """
 
+@wrapt.decorator
+def experiment_test(wrapped_test, instance, args, kwargs):
+    """
+    Convert a LisaTest test method to be automatically called for each experiment
+
+    The method will be passed the experiment object and a list of the names of
+    tasks that were run as the experiment's workload.
+    """
+    for experiment in instance.executor.experiments:
+        tasks = experiment.wload.tasks.keys()
+        try:
+            wrapped_test(experiment, tasks, *args, **kwargs)
+        except AssertionError as e:
+            trace_relpath = os.path.join(experiment.out_dir, "trace.dat")
+            e.args += ("Check trace file: " + os.path.abspath(trace_relpath),)
```

Comment:
OK, pushed a new version that ends up looking like this, which I think is ideal:
```
10:16:15  INFO    : --------------------------------------------------------------------------------
10:16:15  INFO    : 
10:16:15  INFO    : ################################################################################
10:16:15  INFO    :       Executor - Experiments execution completed
10:16:15  INFO    : ################################################################################
10:16:15  INFO    :       Executor - Results available in:
10:16:15  INFO    :       Executor -       /vagrant/results/20161004_101543
Wake Migration: BIG -> LITLLE: 1 ... FAIL
Wake Migration: BIG -> LITLLE: 2 ... ok
Wake Migration: Test First CPU ... ok
Wake Migration: LITTLE -> BIG: 1 ... FAIL
Wake Migration: LITTLE -> BIG: 2 ... ok

======================================================================
FAIL: Wake Migration: BIG -> LITLLE: 1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/lisa/libs/utils/test.py", line 116, in experiment_test
    wrapped_test(experiment, tasks, *args, **kwargs)
  File "/vagrant/tests/eas/acceptance.py", line 396, in test_big_little_switch1
    self._assert_switch(experiment, "little", 0)
  File "/vagrant/tests/eas/acceptance.py", line 376, in _assert_switch
    msg=msg)
AssertionError: Not all tasks wake-migrated to little cores in the expected window: (0.29281700000137789, 1.2928170000013779)
	Check trace file: /vagrant/results/20161004_101543/rtapp::wake_migration/1/trace.dat

======================================================================
FAIL: Wake Migration: LITTLE -> BIG: 1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/lisa/libs/utils/test.py", line 116, in experiment_test
    wrapped_test(experiment, tasks, *args, **kwargs)
  File "/vagrant/tests/eas/acceptance.py", line 381, in test_little_big_switch1
    self._assert_switch(experiment, "big", 1)
  File "/vagrant/tests/eas/acceptance.py", line 376, in _assert_switch
    msg=msg)
AssertionError: Not all tasks wake-migrated to big cores in the expected window: (5.2928170000013779, 6.2928170000013779)
	Check trace file: /vagrant/results/20161004_101543/rtapp::wake_migration/1/trace.dat

----------------------------------------------------------------------
Ran 5 tests in 46.439s

FAILED (failures=2)
```